week3/homework/src/actions.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { readFile, writeFile } = require('fs');
+const { promisify } = require('util');
+const uuid = require('uuid/v4');
+const read = promisify(readFile);
+const write = promisify(writeFile);
+const FILE_PATH = 'todo.json';
+const list = {};
+
+
+const fileContent = (todo) => {
+    return write(FILE_PATH, JSON.stringify(todo, null, 2));
+};
+
+const createTodo = async (todo) => {
+    if (!todo.description) throw new Error('Description is required');
+
+    const todoList = await readTheList();
+    const id = uuid();
+    list[id] = todo;
+    list[id].done = false;
+    todoList.push(list);
+    await fileContent(todoList);
+    return todoList;
+};
+
+const readTheList = () => {
+    return read(FILE_PATH, 'utf-8')
+        .then(JSON.parse)
+        .catch(() => ([]));
+};
+
+const updateTodo = async (id, newTodo) => {
+    const todoList = await readTheList();
+    try {
+        const foundItem = todoList.find(todo => todo[id]);
+        foundItem[id].description = newTodo.description;
+        await fileContent(todoList);
+        return todoList;
+    } catch (error) {
+        throw new Error('Item cannot be found.');
+    }
+};
+
+const markAsDoneOrNotDone = async (id, boolean) => {
```

Comment:
`boolean` is a not good name for a variable since it's not really clear what the intention is. Why not `done`?

Reply:
I agree with George here. `boolean` is less explicit than `done`. The point is to be as specific as possible as to *what the variable actually represents*, rather than its type. The type you can document using JSDoc.

Reply:
ok I will keep that in mind, I just changed that too